### DISCUSSION TITLE
OpenGL canvas (shader) support

### DISCUSSION THREAD

**Discussion by hgomersall** [slint-ui/slint#676] [Ideas]
Very excited by sixtyfps, but our current needs depend on being able to render to an OpenGL context with a series of custom shaders somewhere in the gui stack (currently our system uses QML controls drawn over the top of our shader rendering). I couldn't find anything on this in the current docs, so I assume it's not yet supported. Is support for this in the roadmap?

**Comment by tronical** [comment]
Yeah, this is not supported yet but I'd love to support this. In principle this shouldn't be hard to do, as long as you'd be okay with rendering into an FBO. That is basically tracked in #423 with some details about how it could be implemented.

Or are you looking for another way, i.e. rendering a GUI with sixtyfps into an FBO and integrating that into the rest of your rendering?

**Reply by hgomersall** [reply]
I've been meaning to respond to this for a while, but it looks like the new OpenGL work does what I needed! Great stuff :)

**Reply by tronical** [reply]
Excellent. Let us know if there is anything else missing. Are you okay with just underlay or overlay or do you also need to integrate textures?
